File: Source/Baseline/Iti.Voice/VoiceRecordDto.cs
using System;
using Iti.Baseline.Core.DTOs;
using Iti.Baseline.Identities;

namespace Iti.Baseline.Voice
{
    public class VoiceRecordDto : IDto
    {
        public VoiceRecordId Id { get; set; } = new VoiceRecordId();

        public NotificationId NotificationId { get; set; }

        public VoiceStatus Status { get; set; } = VoiceStatus.Pending;

        public DateTimeOffset? SentUtc { get; set; }

        public string ToAddress { get; set; }

        public string Body { get; set; }

        public int RetryCount { get; set; }
        public DateTimeOffset? NextRetryUtc { get; set; }
    }
}

File: Source/iti-baseline-types/VoiceRecordDto.ts
import { NotificationId } from "./NotificationId";
import { VoiceRecordId } from './VoiceRecordId'
import { VoiceStatus } from './VoiceStatus'



export const VoiceRecordDtoTypeName = 'VoiceRecordDto'
export interface VoiceRecordDto  { 
    id: VoiceRecordId
    notificationId: NotificationId
    status: VoiceStatus
    sentUtc: Date | null| undefined
    toAddress: string
    body: string
    retryCount: number
    nextRetryUtc: Date | null | undefined
}
